Orange/widgets/unsupervised/tests/test_owkmeans.py

```diff
@@ -0,0 +1,43 @@
+from PyQt4 import QtGui
+
+from Orange.widgets.tests.base import WidgetTest
+from Orange.widgets.unsupervised.owkmeans import OWKMeans
+
+from Orange.data import Table
+class TestOWKMeans(WidgetTest):
+
+    def setUp(self):
+        self.widget = self.create_widget(OWKMeans)
+        self.combo_box = self.widget.findChildren(QtGui.QComboBox)
+        self.spinner = self.widget.findChildren(QtGui.QSpinBox)
+        self.radio_button = self.widget.findChildren(QtGui.QRadioButton)
+        self.iris = Table("iris")
+
+    def test_boxes_visible(self):
+        """ Check if all boxes visible """
+        self.assertEqual(self.combo_box[0].isHidden(), False)
+        self.assertEqual(self.combo_box[1].isHidden(), False)
+        self.assertEqual(self.combo_box[2].isHidden(), False)
+        self.assertEqual(self.spinner[0].isHidden(), False)
+        self.assertEqual(self.spinner[1].isHidden(), False)
+        self.assertEqual(self.spinner[2].isHidden(), False)
+
+
+    def test_optimization_report_display(self):
+        """ Check visibility of optimization report after we select the number of clusters  """
+        for button_pos, value in ((0, True),
+                                  (1, False)):
+            self.radio_button[button_pos].click()
+            self.radio_button[button_pos].isChecked()
+            self.assertEqual(self.widget.mainArea.isHidden(), value)
+
+    def test_data_on_output(self):
+        """ Check if data is on output after create widget and run """
+        # Connect iris to widget
+        self.send_signal("Data", self.iris)
+        self.widget.run()
```

Comment:
Rather than call the run method, I would call commit method (or even click the Apply button).